pilot/pkg/networking/util/util.go

```diff
@@ -93,6 +94,8 @@ const (
 	// Envoy Stateful Session Filter
 	// TODO: Move to well known.
 	StatefulSessionFilter = "envoy.filters.http.stateful_session"
+
+	DefaultHTTPSessionHeader = "x-session-affinity"
```

Comment:
how does this work? I thought we needed a header per upstream svc

Reply:
We do not need a separate header per upstream. Currently we can configure using this header (where every service uses the same header to propagate session) which will be replaced at each hop by Envoy if there are multiple hops in the call chain. We can make this configurable later. My preference would be to make this configurable per service  similar to how we did for cookies. That would make this more clearer. But @costinm wanted to make this configurable [later](https://github.com/istio/istio/pull/42592#issuecomment-1381325075)

Reply:
+1 Make it configurable, otherwise it is useless

Reply:
@costinm Are you OK with making this service specific header as well? I think that would be most useful and align with how cookies are working. I can change accordingly if you do not have any concerns.